client/src/plus/collections-quicksearch.ts

```diff
@@ -0,0 +1,73 @@
+const COLLECTION_ITEMS_KEY: string = "collection-items";
+const COLLECTION_ITEMS_UPDATED_DATE_KEY: string =
+  "collection-items-updated-date";
+
+type CollectionItem = {
+  id: number;
+  title: string;
+  url: string;
+};
+
+function getCollectionItemsUpdatedDate(): Date | null {
+  let dateString = window?.localStorage?.getItem(
```

Comment:
The pattern used in https://github.com/mdn/yari/pull/6352 was to wrap all `localStorage` usages in `try...catch`. It allows us to `console.warn()` if it failed, giving users a hint why the feature might not work.